nifi-commons/nifi-utils/src/main/java/org/apache/nifi/stream/io/BufferedInputStream.java

```diff
@@ -16,19 +16,445 @@
  */
 package org.apache.nifi.stream.io;
 
+import java.io.IOException;
 import java.io.InputStream;
 
 /**
  * This class is a slight modification of the BufferedInputStream in the java.io package. The modification is that this implementation does not provide synchronization on method calls, which means
  * that this class is not suitable for use by multiple threads. However, the absence of these synchronized blocks results in potentially much better performance.
  */
-public class BufferedInputStream extends java.io.BufferedInputStream {
+public class BufferedInputStream extends InputStream {
```

Comment:
This is not explicitly public, but I do believe that others are depending on it, so I would be very hesitant to rename it.